provider/provider.go

```diff
@@ -204,6 +204,7 @@ func (p *Provider) GetResources(ctx context.Context) (result map[string]tfsdk.Re
 		"rhcs_rosa_oidc_config_input": &RosaOidcConfigInputResourceType{},
 		"rhcs_rosa_oidc_config":       &RosaOidcConfigResourceType{},
 		"rhcs_dns_domain":             &DNSDomainResourceType{},
+		"rhcs_cluster_autoscaler":     &ClusterAutoscalerResourceType{},
```

Comment:
Having "cluster" included in the name might be confusing for user, as it might cause the user it related to cluster creation.
I understand that you added the "cluster" to make it clear that the autoscaler belongs to the cluster.
@arendej any thoughts here?

Reply:
naming it autoscaler would be ambiguous in my opinion

Reply:
I agree with @osherdp here. But ill wait for @arendej response.

Reply:
per discussion, we can leave the name as-is. 
@EricPonvelle please point out in documentation about rhcs_cluster_autoscaler about how it is scoped to worker machinepools and not the control plane/infra.